crates/solidity/outputs/cargo/crate/src/generated/language.rs

```diff
@@ -2940,114 +1908,60 @@ impl Language {
+                let result = parse_left_equality_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_comparison_expression(input);
+                let result = parse_left_comparison_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_comparison_expression(input);
+                let result = parse_left_bitwise_or_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_comparison_expression(input);
+                let result = parse_left_bitwise_xor_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_comparison_expression(input);
+                let result = parse_left_bitwise_and_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_bitwise_or_expression(input);
+                let result = parse_left_shift_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_bitwise_xor_expression(input);
+                let result = parse_left_additive_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_bitwise_and_expression(input);
+                let result = parse_left_multiplicative_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_shift_expression(input);
+                let result = parse_left_exponentiation_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_shift_expression(input);
+                let result = parse_right_exponentiation_expression(input);
                 choice.consider(input, result)?;
-                let result = parse_shift_expression(input);
-                choice.consider(input, result)?;
-                let result = parse_additive_expression(input);
-                choice.consider(input, result)?;
-                let result = parse_additive_expression(input);
-                choice.consider(input, result)?;
-                let result = parse_multiplicative_expression(input);
-                choice.consider(input, result)?;
-                let result = parse_multiplicative_expression(input);
-                choice.consider(input, result)?;
-                let result = parse_multiplicative_expression(input);
-                choice.consider(input, result)?;
-                if !self.version_is_at_least_0_6_0 {
-                    let result = parse_exponentiation_expression_removed_from_0_6_0(input);
-                    choice.consider(input, result)?;
-                }
-                if self.version_is_at_least_0_6_0 {
-                    let result = parse_exponentiation_expression_introduced_from_0_6_0(input);
-                    choice.consider(input, result)?;
-                }
                 choice.finish(input)
```

Comment:
before this, we discarded operators that don't match the current version. But now we consider all of them in the `choice`, and inside it, we unconditionally create another `choice`, and check for the version inside.

I wonder if we should prune it out beforehand?

Reply:
I've stopped unconditionally emitting a choice where it's not necessary. See https://github.com/NomicFoundation/slang/pull/699#discussion_r1421411123 for more context.